src/AutoMapper/Mappers/ExpressionMapper.cs

```diff
@@ -247,7 +247,13 @@ private Type GetSourceType(PropertyMap propertyMap)
             {
                 if (propertyMap.SourceMember == null)
                 {
-                    return propertyMap.SourceType;
+                    return propertyMap.SourceType ??
```

Comment:
Yea I'm not 100% sure what would happen, but it might fail somewhere else.  Not sure if UseAsDataSource would work with a value resolver, but I would guess it wouldn't since it's similar to projection in what it does.  Well even if it did projection wouldn't support it anyways :)